--- rpcbind-0.1.4/src/Makefile.am~	2004-11-15 14:38:02.000000000 +0100
+++ rpcbind-0.1.4/src/Makefile.am	2007-04-24 16:29:08.000000000 +0200
@@ -1,7 +1,4 @@
-INCLUDES = -I$(srcdir)/tirpc -DPORTMAP -DINET6 -DVERSION="\"$(VERSION)\"" \
-                                                -D_GNU_SOURCE -Wall -pipe
-
-bin_PROGRAMS = rpcbind rpcinfo
+sbin_PROGRAMS = rpcbind rpcinfo
 
 rpcbind_SOURCES =       check_bound.c rpcbind.c \
                         rpcb_svc_4.c rpcb_svc_com.c \
@@ -14,10 +11,12 @@
 rpcinfo_LDADD   =       $(LIB_TIRPC)
 
 
-rpcbind_LDFLAGS = -lpthread -ltirpc
+rpcbind_LDFLAGS = -lpthread -ltirpc -lwrap
 rpcbind_LDADD = $(LIB_TIRPC)
-AM_CPPFLAGS = -I/usr/include/tirpc -DCHECK_LOCAL -DPORTMAP \
-                       -DFACILITY=LOG_MAIL -DSEVERITY=LOG_INFO
+
+AM_CFLAGS = -I/usr/include/tirpc -DCHECK_LOCAL -DPORTMAP -DINET6 -DLIBWRAP \
+                       -DFACILITY=LOG_AUTH -DSEVERITY=LOG_INFO -DWARMSTART \
+		       -DVERSION="\"$(VERSION)\"" -D_GNU_SOURCE -Wall -pipe
 ##                     -DRPCBIND_DEBUG -DSVC_RUN_DEBUG -DDEBUG_RMTCALL \
 ##                     -DND_DEBUG -DBIND_DEBUG
 
