--- rpcbind-1.2.5/systemd/rpcbind.service.in.orig	2018-08-26 11:29:00.657170237 +0200
+++ rpcbind-1.2.5/systemd/rpcbind.service.in	2018-08-26 21:41:05.066750909 +0200
@@ -8,11 +8,13 @@
 # rpcbind.socket, no matter how this unit is started.
 Requires=rpcbind.socket
 Wants=rpcbind.target
+After=syslog.target network.target
 
 [Service]
 Type=notify
-# distro can provide a drop-in adding EnvironmentFile=-/??? if needed.
+EnvironmentFile=-/etc/sysconfig/rpcbind
 ExecStart=@_sbindir@/rpcbind $RPCBIND_OPTIONS -w -f
 
 [Install]
 WantedBy=multi-user.target
+Also=rpcbind.socket
--- rpcbind-0.2.4/systemd/rpcbind.socket.orig	2016-11-28 21:47:28.000000000 +0100
+++ rpcbind-0.2.4/systemd/rpcbind.socket	2018-04-11 19:43:27.366604185 +0200
@@ -5,14 +5,14 @@
 Before=rpcbind.target
 
 [Socket]
-ListenStream=/run/rpcbind.sock
+ListenStream=/var/run/rpcbind.sock
 
 # RPC netconfig can't handle ipv6/ipv4 dual sockets
-BindIPv6Only=ipv6-only
-ListenStream=0.0.0.0:111
-ListenDatagram=0.0.0.0:111
-ListenStream=[::]:111
-ListenDatagram=[::]:111
+#BindIPv6Only=ipv6-only
+#ListenStream=0.0.0.0:111
+#ListenDatagram=0.0.0.0:111
+#ListenStream=[::]:111
+#ListenDatagram=[::]:111
 
 [Install]
 WantedBy=sockets.target
