--- rpcbind-1.2.7/src/rpcbind.c.orig	2024-10-05 21:02:24.365984122 +0200
+++ rpcbind-1.2.7/src/rpcbind.c	2024-10-05 21:07:07.604449687 +0200
@@ -341,7 +341,7 @@ init_transport(struct netconfig *nconf)
 	int status;	/* bound checking ? */
 	int aicode;
 	int addrlen = 0;
-	int nhostsbak;
+	int nhostsbak = 1;
 	int checkbind;
 	int on = 1;
 	struct sockaddr *sa = NULL;
@@ -458,8 +458,6 @@ init_transport(struct netconfig *nconf)
 		hints.ai_family = si.si_af;
 		hints.ai_socktype = si.si_socktype;
 		hints.ai_protocol = si.si_proto;
-	}
-	if (nconf->nc_semantics == NC_TPI_CLTS) {
 		/*
 		 * If no hosts were specified, just bind to INADDR_ANY.  Otherwise
 		 * make sure 127.0.0.1 is added to the list.
@@ -477,21 +475,15 @@ init_transport(struct netconfig *nconf)
 			} else
 				return 1;
 		}
-
-	       /*
-		* Bind to specific IPs if asked to
-		*/
-		checkbind = 0;
-		while (nhostsbak > 0) {
-			--nhostsbak;
-			/*
-			 * XXX - using RPC library internal functions.
-			 */
-			if ((fd = __rpc_nconf2fd(nconf)) < 0) {
-				syslog(LOG_ERR, "cannot create socket for %s",
-				    nconf->nc_netid);
-				return (1);
-			}
+	}
+       /*
+	* Bind to specific IPs if asked to
+	*/
+	checkbind = 0;
+	while (nhostsbak > 0) {
+		--nhostsbak;
+		if ((strcmp(nconf->nc_netid, "local") != 0) &&
+		    (strcmp(nconf->nc_netid, "unix") != 0)) {
 
 			hints.ai_flags &= ~AI_NUMERICHOST;
 			switch (hints.ai_family) {
@@ -520,7 +512,7 @@ init_transport(struct netconfig *nconf)
 					    host_addr) == 1)
 						continue;
 				}
-	        		break;
+				break;
 			default:
 				break;
 			}
@@ -543,105 +535,41 @@ init_transport(struct netconfig *nconf)
 			}
 			addrlen = res->ai_addrlen;
 			sa = (struct sockaddr *)res->ai_addr;
-			oldmask = umask(S_IXUSR|S_IXGRP|S_IXOTH);
-                        if (bind(fd, sa, addrlen) != 0) {
-				syslog(LOG_ERR, "cannot bind %s on %s: %m",
-					(hosts[nhostsbak] == NULL) ? "*" :
-					hosts[nhostsbak], nconf->nc_netid);
-				if (res != NULL) {
-					freeaddrinfo(res);
-					res = NULL;
-				}
-				continue;
-			} else
-				checkbind++;
-			(void) umask(oldmask);
-
-			/* Copy the address */
-			taddr.addr.maxlen = taddr.addr.len = addrlen;
-			taddr.addr.buf = malloc(addrlen);
-			if (taddr.addr.buf == NULL) {
-				syslog(LOG_ERR,
-				    "cannot allocate memory for %s address",
-				    nconf->nc_netid);
-				if (res != NULL)
-					freeaddrinfo(res);
-				return 1;
-			}
-			memcpy(taddr.addr.buf, sa, addrlen);
-#ifdef RPCBIND_DEBUG
-			if (debugging) {
-				/*
-				 * for debugging print out our universal
-				 * address
-				 */
-				char *uaddr;
-				struct netbuf nb;
-				int sa_size = 0;
-
-				nb.buf = sa;
-				switch( sa->sa_family){
-				case AF_INET:
-				  sa_size = sizeof (struct sockaddr_in);
-				  break;
-				case AF_INET6:
-				  sa_size = sizeof (struct sockaddr_in6);				 
-				  break;
-				}
-				nb.len = nb.maxlen = sa_size;
-				uaddr = taddr2uaddr(nconf, &nb);
-				(void) xlog(LOG_DEBUG,
-				    "rpcbind : my address is %s\n", uaddr);
-				(void) free(uaddr);
-			}
-#endif
-			my_xprt = (SVCXPRT *)svc_tli_create(fd, nconf, &taddr, 
-                                RPC_MAXDATASIZE, RPC_MAXDATASIZE);
-			if (my_xprt == (SVCXPRT *)NULL) {
-				syslog(LOG_ERR, "%s: could not create service", 
-                                        nconf->nc_netid);
-				goto error;
-			}
 		}
-		if (!checkbind)
-			return 1;
-	} else {	/* NC_TPI_COTS */
+
+		/*
+		 * XXX - using RPC library internal functions.
+		 */
 		if ((fd = __rpc_nconf2fd(nconf)) < 0) {
 			syslog(LOG_ERR, "cannot create socket for %s",
 			    nconf->nc_netid);
-			return (1);
-		}
-
-		if ((strcmp(nconf->nc_netid, "local") != 0) &&
-		    (strcmp(nconf->nc_netid, "unix") != 0)) {
-			if ((aicode = getaddrinfo(NULL, servname, &hints, &res))!= 0) {
-			  if ((aicode = getaddrinfo(NULL, "portmapper", &hints, &res))!= 0) {
-			  printf("cannot get local address for %s: %s",  nconf->nc_netid, gai_strerror(aicode));
-			  syslog(LOG_ERR,
-				    "cannot get local address for %s: %s",
-				    nconf->nc_netid, gai_strerror(aicode));
-				return 1;
-			  }
-			}
-			addrlen = res->ai_addrlen;
-			sa = (struct sockaddr *)res->ai_addr;
+			continue;
 		}
 		oldmask = umask(S_IXUSR|S_IXGRP|S_IXOTH);
-		__rpc_fd2sockinfo(fd, &si);
-		if (setsockopt(fd, SOL_SOCKET, SO_REUSEADDR, &on,
-				sizeof(on)) != 0) {
-			syslog(LOG_ERR, "cannot set SO_REUSEADDR on %s",
-				nconf->nc_netid);
-			if (res != NULL)
-				freeaddrinfo(res);
-			return 1;
+		if (nconf->nc_semantics != NC_TPI_CLTS) {
+			__rpc_fd2sockinfo(fd, &si);
+			if (setsockopt(fd, SOL_SOCKET, SO_REUSEADDR, &on,
+					sizeof(on)) != 0) {
+				syslog(LOG_ERR, "cannot set SO_REUSEADDR on %s",
+					nconf->nc_netid);
+				if (res != NULL) {
+					freeaddrinfo(res);
+					res = NULL;
+				}
+				continue;
+			}
 		}
 		if (bind(fd, sa, addrlen) < 0) {
-			syslog(LOG_ERR, "cannot bind %s: %m", nconf->nc_netid);
-			if (res != NULL)
+			syslog(LOG_ERR, "cannot bind %s on %s: %m",
+				(hosts && hosts[nhostsbak]) ? hosts[nhostsbak] : "*",
+				nconf->nc_netid);
+			if (res != NULL) {
 				freeaddrinfo(res);
-			return 1;
-		}
+				res = NULL;
+			}
+			continue;
+		} else
+			checkbind++;
 		(void) umask(oldmask);
 
 		/* Copy the address */
@@ -660,7 +586,7 @@ init_transport(struct netconfig *nconf)
 			/* for debugging print out our universal address */
 			char *uaddr;
 			struct netbuf nb;
-		        int sa_size2 = 0;
+			int sa_size2 = 0;
 
 			nb.buf = sa;
 			switch( sa->sa_family){
@@ -679,7 +605,8 @@ init_transport(struct netconfig *nconf)
 		}
 #endif
 
-		listen(fd, SOMAXCONN);
+		if (nconf->nc_semantics != NC_TPI_CLTS)
+			listen(fd, SOMAXCONN);
 
 		my_xprt = (SVCXPRT *)svc_tli_create(fd, nconf, &taddr, RPC_MAXDATASIZE, RPC_MAXDATASIZE);
 		if (my_xprt == (SVCXPRT *)NULL) {
@@ -744,6 +671,8 @@ got_socket:
 		/* Also add version 2 stuff to rpcbind list */
 		rbllist_add(PMAPPROG, PMAPVERS, nconf, &taddr.addr);
 	}
+	if (!checkbind)
+		return 1;
 
 	/* We need to support portmap over IPv4. It makes sense to
 	 * support it over AF_LOCAL as well, because that allows
