--- rpcbind-1.2.6/src/rpcbind.c.orig	2022-05-07 20:58:18.116315820 +0200
+++ rpcbind-1.2.6/src/rpcbind.c	2022-05-07 20:58:43.826176537 +0200
@@ -132,7 +132,7 @@ char *tcptrans;		/* Name of TCP transpor
 char *udp_uaddr;	/* Universal UDP address */
 char *tcp_uaddr;	/* Universal TCP address */
 #endif
-static char servname[] = "rpcbind";
+static char servname[] = "sunrpc";
 static char superuser[] = "superuser";
 
 int main(int, char *[]);
--- rpcbind-0.1.7/src/rpcinfo.c~	2010-01-08 16:14:24.592156602 +0000
+++ rpcbind-0.1.7/src/rpcinfo.c	2010-01-08 16:14:31.578838609 +0000
@@ -1842,7 +1842,7 @@
 
   /* Get the address of the rpcbind */
   memset (&hints, 0, sizeof hints);
-  if ((getaddrinfo (host, "rpcbind", &hints, &res) != 0) &&
+  if ((getaddrinfo (host, "sunrpc", &hints, &res) != 0) &&
       (getaddrinfo (host, "portmapper",&hints, &res) != 0))
     {
       rpc_createerr.cf_stat = RPC_N2AXLATEFAILURE;
